src/Instrument/Transformer/CachingTransformer.php

```diff
@@ -99,6 +109,9 @@ public function transform(StreamMetaData $metadata)
             }
             $this->processTransformers($metadata);
             file_put_contents($cacheUri, $metadata->source);
+            if (!$cacheExisted && $this->cachePerms){
+                chmod($cacheUri, $this->cachePerms);
```

Comment:
No need to check `$this->cachePerms`, just always change to the filemode to the specific. Suggestion: rename `cachePerms` to the `cacheFileMode`